### ISSUE TITLE
Does it work with new Xiaomi Smartmi Pure Evaporative Air Humidifier

### ISSUE THREAD

**Issue by ghost** [acc-ua/homebridge-xiaomi-humidifier#13]

**Comment by leoneleone** [comment]
@malmentm 
This plugin does not. 
I’ve forked another plugin and have a branch named “humidifier.ca1”. That branch will work with the new SmartMi Humidifier you mention. 

Have you managed to extract the token for your device yet? That’s the hardest part of the process with the new Humidifiers.

**Comment by ghost** [comment]
@leoneleone no i have not becausse i don't own one yet. I guess it's the same procedure as for the Ceiling Lamp i have, exctract from APP DB. I wanted to know if i could get it working with homekit beforde i order one :)

**Comment by leoneleone** [comment]
@malmentm 
Yes you can get it working with HomeKit, and it works quite well 🙂 

Extracting the app db is the way to go.
the best guide to use for the process:
https://github.com/jghaanstra/com.xiaomi-miio/blob/master/docs/obtain_token.md

The plugin you want to use is this:
https://github.com/russtone/homebridge-mi-humidifier

If you want full functionality of the new SmartMi Humidifier (zhimi.humidifier.ca1), please use this specific branch of my fork:
https://github.com/leoneleone/homebridge-mi-humidifier/tree/leoneleone-humidifier.ca1

**Comment by ghost** [comment]
@leoneleone OK nice, then i can order the Xiaomi Smartmi Pure Evaporative  and try to get it working in homebridge :D Does it show up as humidifier in HomeApp of a Fan with the new fork?

**Comment by leoneleone** [comment]
@malmentm 
Shows up as a Humidifier.  Please read the notes I’ve made in the index.js file of that fork. 

One thing I’d like to point out:
The SmartMi Evaportive Humidifier has a Target Humidity set to 70%. This is hard coded and the iOS MiHome.app has no way of changing that setting, and there is no way to set this manually either. 

Fortunately, you will be able to control that property with the fork mentioned using a 3rd Party HomeKit app (like Elgatos Eve.app), or using Siri Voice Commands, but that Characteristic (Target Humidity) is not yet supported for humidifiers by the native Home.app

**Comment by Malevich2x2** [comment]
I can't do it to work, homekit doesn't want to add this device

**Comment by Malevich2x2** [comment]
But how to install this particular branch, which command to register in the terminal https://github.com/leoneleone/homebridge-mi-humidifier/tree/leoneleone-humidifier.ca1
